### PR TITLE
Lto1 internal compiler error

### PR DESCRIPTION
**Pull request by praasz**
### Details:
 - *Fix LTO issue*
 - *NodeRegistry class used by MatcherPass class*

### Tickets:
 - *90386*

### CODE REVIEW

File: src/core/include/openvino/pass/graph_rewrite.hpp [hunk 1]
```diff
@@ -17,6 +17,52 @@ using graph_rewrite_callback = std::function<bool(pass::pattern::Matcher& m)>;
 using recurrent_graph_rewrite_callback = std::function<bool(pass::pattern::RecurrentMatcher& m)>;
 using handler_callback = std::function<bool(const std::shared_ptr<Node>& node)>;
 namespace pass {
+/// \brief Register openvino node pointers into container.
+/// Can create and/or add existing node pointers into register
+class NodeRegistry {
```
Comment (ilyachur) [inline src/core/include/openvino/pass/graph_rewrite.hpp:22]: Why did you move this class from developer to public API?

Comment (praasz) [reply src/core/include/openvino/pass/graph_rewrite.hpp:22]: To re-use it `graph_rewrite.hpp` header.
Developer API is not visible in this header.

**Review by ilyachur — APPROVED** [review]
